### DISCUSSION TITLE
Consider BookBrainz as a future data provider

### DISCUSSION THREAD

**Discussion by ghost** [mateusz-bak/openreads#701] [Ideas]
BookBrainz is a new user-based book database with a clean interface and an active team, which could be very useful as an alternative data provider in the app.
For example, it'd be nice if the user could choose between searching on OpenLibrary, BookBrainz, or both.

Bookbrainz website: https://bookbrainz.org/

BookBrainz is part of a bigger foundation called MetaBrainz: https://metabrainz.org/

**Comment by mateusz-bak** [comment]
Hey @felysu thanks for the suggestion! Will take a look at this source and check if it fits with the app.

**Comment by mizzunet** [comment]
Would you add Google Book's API as an option if user wants to enable?

I'm aware of the fact that this may not be privacy respecting. I think, some user would attract if it has Google, as OpenLibrary is not that good source.

https://developers.google.com/books/docs/v1/using

Edit:
or as Goodreads doens't provide API anymore, I'll try to see if I can make a custom API for Goodreads

**Comment by mizzunet** [comment]
@mateusz-bak if you're okay with Goodreads, here's an API thing, https://mizzunet.eu.org/goodreads/. It's not perfect but it works.

'Simple' scrapes data from search results, so it's faster.
'More' visits each result page then scrapes from there.

Edit:

I guess this may be used for cover images

Edit 2:
Issue opened https://github.com/mateusz-bak/openreads-android/issues/110

**Comment by edern-appere** [comment]
Hi everyone !

I also think it would be great to have multiple sources.

I've been using Openreads lately as I'm looking for a replacement of the closed-sourced app I've used for quite a while to manage my readings ([Mylibrary](https://play.google.com/store/apps/details?id=com.vgm.mylibrary&gl=US)).
Openreads has an awesome user interface and feature set but the results from the OpenLibrary database are often quite deceiving (especially for non-english books). I think it would be great to have multiple source to find books. 

I thought about two solutions :
- have optional databases (user enabled in the settings)
- run book searches first on OpenLibrary and if nothing's found, look for results in other databases

I don't know exactly which databases Mylibrary uses, but the results are much more relevant (I'm wondering if it scrapes data from amazon...).

**Comment by ropery** [comment]
https://isbndb.com seems large/useful too.

**Comment by apobrt** [comment]
> https://isbndb.com seems large/useful too.

It seems proprietary and not free

**Comment by apobrt** [comment]
> BookBrainz is a new user-based book database with a clean interface and an active team, which could be very useful as an alternative data provider in the app.

It is still in a beta stage, under development, but why not?

> Would you add Google Book's API as an option if user wants to enable?

Usually I prefer not to use proprietary services and instead support open source ones (like OpenLibrary or BookBrainz), but maybe adding this feature as disabled by default, as an opt-in with very clear alert to the end user could be sensible and useful.

**Comment by x51mon** [comment]
I agree with @apobrt about a non-free data provider. I prefer using free and open-source one but I've personally stopped using the app because I found it to annoying having to enter almost every book manually.

**Comment by edern-appere** [comment]
I'm in the same situation : I prefer to rely on free and open-source software but if I cannot find my books half the time with open databases, it's anoying and I no longer use the app.
   
  
    
   
  
   Actually I still use an old-looking and non-free app to manage my books (https://play.google.com/store/apps/details?id=com.vgm.mylibrary) because it always finds the books i'm looking for (i think it scraps data from various sources).
   
   
   
    Le 04/09/2023 21:53 CEST, x51mon ***@***.***> a écrit :
    
   I agree with @apobrt about a non-free data provider. I prefer using free and open-source one but I've personally stopped using the app because I found it to annoying having to enter almost every book manually.

**Comment by mateusz-bak** [comment]
Thanks for all the suggestions. I won't be adding any proprietary sources at this stage.

I will take a look at BookBrainz now and see if it can be implemented.

**Comment by leoossa** [comment]
@mateusz-bak 
Hi,
I know it's been a long time since this issue was opened.
Though I think I found quite neat acceptable solution you can consider.
There's quite nice API here:
https://data.bn.org.pl/networks/bibs
https://www.bn.org.pl/aktualnosci/3345-biblioteka-narodowa-otwiera-najwieksza-polska-baze-danych-bibliograficznych.html

For example - I couldn't find this book by ISBN: 9788375611731 using current OpenLibrary search in the app, but I was able to find it via BN search.

They claim it's the biggest database in Poland:

```
Katalog Biblioteki Narodowej jest największą i najpełniejszą bazą danych bibliograficznych w Polsce – 4 678 549 rekordów bibliograficznych i 2 175 147 rekordów wzorcowych
```


and the API it's quite easy to consume.
